src/main/java/alpaca/tasks/Task.java

```diff
@@ -0,0 +1,95 @@
+package alpaca.tasks;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public abstract class Task {
+    private String name;
+    private Boolean done;
+    private String type;
+    private ArrayList<LocalDate> dates;
+
+    protected Task(String name) {
+        this.name = name;
+        this.done = false;
+        processDates();
+        processParams();
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public Boolean isDone() {
+        return this.done;
+    }
+
+    public void setDone() {
+        this.done = true;
+    }
+
+    public void setUndone() {
+        this.done = false;
+    }
+
+    protected void setType(String type) {
+        this.type = type;
+    }
+
+    private void processDates() {
+        this.dates = new ArrayList<>();
+        Pattern pattern = Pattern.compile("\\d{4}-\\d{1,2}-\\d{1,2}|\\d{1,2}-\\d{1,2}-\\d{4}|\\d{4}\\/\\d{1,2}\\/\\d{1,2}|\\d{1,2}\\/\\d{1,2}\\/\\d{4}");
+        Matcher matcher = pattern.matcher(this.name);
+        while (matcher.find()) {
+            try {
+                String dateStr = matcher.group();
+                String dateMod = dateStr.replaceAll("\\/", "-");
+                dateMod = dateMod.replaceFirst("(^|[^\\d])(\\d)($|[^\\d])", "$10$2$3");
+                dateMod = dateMod.replaceFirst("(^|[^\\d])(\\d)($|[^\\d])", "$10$2$3");
+                dateMod = dateMod.replaceAll("(\\d{2})-(\\d{2})-(\\d{4})", "$3-$2-$1");
```

Comment:
Might be a little hard to read the regex, consider adding JavaDocs or use something like
```java
protected static final Pattern FORMATTER = Pattern.compile("(.+) /by (.+)");
Matcher matcher = Deadline.FORMATTER.matcher(input);
String name = matcher.group(1);
String byStr = matcher.group(2);
```